tuning/wmma-contraction.jl

```diff
@@ -0,0 +1,211 @@
+## constants
+using JSON
+
+const N_vals = 2 .^ (7:14)
+
+const data_type = Float16
+const compute_type = Float16
+const accumulate_type = Float32
+
+const zero_c = true
+
+config_path = joinpath(@__DIR__, "../test/benchmark-suite.json")
+fp = open(config_path, "r")
+const jsonData = JSON.parse(read(fp, String))
+
+const MEMORY_USAGE = maximum(N_vals)^2 * 2 * sizeof(data_type) +
+                     maximum(N_vals)^2 * 2 * sizeof(data_type)
```

Comment:
I'm seeing _lots_ of soft OOMs (i.e., not because of the device being full, but because of running against the configured memory limit). Are you sure this is correct, and that your code doesn't perform any additional allocations?

```
┌ Info: 2024-02-26 07:46:27 Not enough memory for configuration 1.2.3.4-5.1.6.3-2.6.4.5 (64×256×128 block, 2×2 warp, 32×8×16 operator, pipelined kernel)
│ Out of GPU memory trying to allocate 648.000 MiB
│ Effective GPU memory usage: 6.56% (3.105 GiB/47.298 GiB)
│ Memory pool usage: 1.898 GiB (2.125 GiB reserved)
│ Memory limit: hard = 2.125 GiB
```

Reply:
Is `N_vals` even used for the tensor contractions? I thought @wardvermeulen used the extents from the TTCG suite, in `test/benchmark-suite.json`, so I think the memory usage estimate given here is completely wrong.